observations/healthy-to-healthy-partially-scaled.md

```diff
@@ -0,0 +1,54 @@
+# What happens when deploying a 'good' build when the service is already partially scaled up?
+
+In this test we went from [application version](../dist) `ABC` to `XYZ` in the stack:
+- [`ScalingAsgRollingUpdate`](../packages/cdk/lib/scaling-asg-rolling-update.ts) (CFN stack `playground-CODE-scaling-asg-rolling-update`)
+
+The main aim of this test was to establish whether deploying whilst a service is partially scaled up works as desired.
+
+## Highlights
+
+The current implementation leads to a temporary scale down during deployment; this is undesirable!
+
+Some potential options for mitigating this problem are discussed below.
+
+## Timeline
+
+1. [Build number 79 was deployed](https://riffraff.gutools.co.uk/deployment/view/322bf36b-fd4d-4fe4-bf72-ce56bb789a08) (in order to start the test from a clean state - running build `ABC`)
+2. Artificial traffic was sent to `https://scaling.rolling-update.gutools.co.uk/healthcheck` in order to trigger a scale-up event
+3. The service scales up to 6 instances (from 3)
+4. [Build number 80 was deployed](https://riffraff.gutools.co.uk/deployment/view/da3d810c-d055-4765-9602-141233e82b45) (which updates to build `XYZ`)
+5. The CFN stack `playground-CODE-scaling-asg-rolling-update` started updating:
+
+   First:
+   > Rolling update initiated. Terminating 6 obsolete instance(s) in batches of 6, while keeping at least 3 instance(s) in service. Waiting on resource signals with a timeout of PT5M when new instances are added to the autoscaling group.
+   
+   Then the ASG capacity is updated:
+   > Temporarily setting autoscaling group MinSize and DesiredCapacity to 9.
+
+7. Once three `SUCCESS` signals are received, six instances are terminated. _At this point we are under-provisioned._
```

Comment:
Think this should be point `6.`?

Reply:
> Think this should be point `6.`?

Yes, my mistake - thanks for catching this